src/Data/HashTable/ST/Basic.hs

```diff
@@ -620,6 +681,12 @@ recordIsDeleted = (== deletedMarker)
 
 
 ------------------------------------------------------------------------------
+{-# INLINE recordIsFilled #-}
+recordIsFilled :: Elem -> Bool
+recordIsFilled el = not (recordIsEmpty el) && not (recordIsDeleted el)
```

Comment:
Oh, ok, I didn't know that `/=#` hasn't been implemented until recently.

However, from what I understand, `hashtables` won't work with `GHC < 6.10` anyways because `hashtables` requires `base == 4.*` (`GHC-6.10.1` is the first version to use `base-4.0.0.0`). And GHC versions `>= 6.10` do have `/=#` implemented ( https://downloads.haskell.org/~ghc/6.10.1/docs/html/libraries/ghc-prim/GHC-Prim.html#3 ). It seems to me that that the compatibility wouldn't really be affected by this piece of code. Unless, again, I'm missing something.